File: Leetcode/Python/Q58_Length_of_Last_Word.py
class Solution(object):
    def lengthOfLastWord(self, s):
        """
        :type s: str
        :rtype: int
        """
        # Split the string into a list of words using whitespace as the delimiter
        s = s.split()
        
        # Return the length of the last word in the list
        # s[len(s)-1] accesses the last word, and len() gets its length
        return len(s[len(s)-1])

File: Leetcode/Java/Q58_Length_of_Last_Word.java
class Solution {
    public int lengthOfLastWord(String s) {
        // Replace multiple spaces with a single space
        s = s.replaceAll("\\s+", " ");
        
        // Split the string into an array of words using space as the delimiter
        String strArray[] = s.split(" ");
        
        // Return the length of the last word in the array
        // strArray[strArray.length-1] accesses the last element of the array
        // .length() gets the length of that last word
        return strArray[strArray.length-1].length();
    }
}
